### PR TITLE
LPS-192372 | Allow users to define endpoints for objects with site scope

### PR DESCRIPTION
**Pull request by Hijky**
Background:
Add a testcase to allow users to define endpoints for objects with site scope

Test Plan
CannotSeeUnpublishedAPIApplicationWithSiteScopedEndpoint
Given with postAPIApplication to create an unpublished api Application with site scope endpoint
When I navigate to /o/api
Then c/{apiApplication} is not present under REST Applications

CanSeePublishedAPIApplicationWithSiteAndCompanyScopedEndpointsWithoutSchemas
Given with postAPIApplication to create a published api Application with site scope endpoint and company scope endpoint
When I navigate to /o/c/{apiApplication}
Then getScopes{siteScopeEndpointPath}Page is present under default
And Then get{companyScopeEndpointPath}Page is present under default

CanExecuteGetSiteScopeEndpointInAPIExplorer
Given with postAPIApplication to create a published api Application with site scope endpoint with related schema
When I execute getScopes{endpointPath}Page with {scopyKey} in /o/c/{apiApplication}
Then I can get correct 200 response code

CanSeePublishedAPIApplicationWithSiteAndCompanyScopedEndpointsWithSchemas
Given created an published api application with site scope endpoint and company scope endpoint with related request and response schemas
When I navigate to /o/c/{apiApplication}
Then getScopeScopeKey{siteScopeEndpointPath}Page is present under {schemaName}
And Then get{companyScopeEndpointPath}Page is present under {schemaName}

Jira ticket:
https://liferay.atlassian.net/browse/LPS-194591

### CODE REVIEW

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

**Comment by Hijky** [comment]
The failures are not related to PR

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/macros/ApplicationAPI.macro [hunk 1]
```diff
@@ -39,7 +39,7 @@ definition {
 		''';
 
 		if (isSet(relatedEndpoint)) {
-			if (!(isSet(ScopeKey))) {
+			if (!(isSet(scopeKey))) {
```
Comment (yanan) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/macros/ApplicationAPI.macro:42]: :+1:

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationEndpoints/AllowUsersToDefineEndpointsForObjectsWithSiteScope.testcase [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+				endpointId = ${endpointId1},
+				requestSchemaId = ${schemaId1});
+
+			SchemaAPI.relateResponseSchemaToEndpointByIds(
+				endpointId = ${endpointId1},
+				responseSchemaId = ${schemaId1});
+
+			var apiApplicationId = JSONPathUtil.getIdValue(response = ${responseFromApplication});
+
+			var responseFromEndpoint = EndpointAPI.createAPIEndpoint(
```
Comment (yanan) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationEndpoints/AllowUsersToDefineEndpointsForObjectsWithSiteScope.testcase:184]: We can check how to improve to create application with multiple endpoints in the future, we're good for now, thanks!

**Review by yanan — COMMENTED** [review]
@Hijky All good, the only thing want to make sure is: could you help remove the properties and see if it could run ok on CI? 
```
property test.liferay.virtual.instance = "false";
property test.run.type = "single";
```
I think at least we should remove the test.run.type=single since our tests don't require this.

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

**Comment by yanan** [comment]
ci:test:headless-functional

**Comment by Hijky** [comment]
The failures are not related to PR

**Review by yanan — APPROVED** [review]

**Review by magjed4289 — APPROVED** [review]

**Pull request closed without merging**